docs/source/contributing/testing.md

```diff
@@ -1,90 +1,163 @@
+The following guidelines for writing tests using Vitest will help you migrate your tests from Jest.
+Vitest shares a similar syntax with Jest, as both use Mocha/Chai, but there are notable differences in handling mocks and other features.
 
-```shell
-RAZZLE_JEST_CONFIG=my-custom-jest-config.js pnpm test
+Similar to Jest, Vitest provides functions such as `it`, `expect`, `describe`, `test`, and `vi`.
+These properties are globally declared in the {file}`test-setup-globals.js` file, making them available throughout the Volto core without requiring explicit imports in individual test files.
+
+
+### Differences in mocks
+
+```javascript
+jest.mock('../../manage/Workflow/Workflow', () =>
+  jest.fn(() => <div id="state-select" />),
+);
 ```
 
-```{note}
-Both configurations are merged in a way that the keys of the config provided override  the initial (`package.json`) default config, either in Volto or in your projects.
+The Vitest equivalent mock to the above Jest mock is the following.
+
+```javascript
+vi.mock('../../manage/Workflow/Workflow', () => ({
+  default: vi.fn(() => <div id="state-select" />),
+}));
+```
+
+Vitest's `vi.mock()` does not automatically assume a default export like Jest does.
+Instead, it treats the module as an object, so the mocked function must be explicitly assigned to the `default` property.
+For more details, refer to the [Vitest Mocking Guide](https://vitest.dev/guide/mocking.html).
+
+
+### Testing with lazy loaded libraries
+
+In Jest, you would test lazy loaded libraries as shown.
+
+```javascript
+jest.mock('@plone/volto/helpers/Loadable/Loadable');
+beforeAll(
+  async () =>
+    await require('@plone/volto/helpers/Loadable/Loadable').__setLoadables(),
+);
+```
+
+In Vitest the equivalent is the following example.
+
+```javascript
+vi.mock('@plone/volto/helpers/Loadable/Loadable');
+beforeAll(async () => {
+  const { __setLoadables } = await import('@plone/volto/helpers/Loadable/Loadable');
+  await __setLoadables();
+});
 ```
 
-This is especially useful in GitHub workflows while developing add-ons.
-You can pass a specific configuration file that properly deals with the add-on configuration.
+`require()` is not supported in Vitest.
+Instead, use `import()`.
+
+
+### Complete migration guide from Jest to Vitest
+You can refer {doc}`../development/add-ons/test-add-ons-18` for more migration information.
+For complete details on migrating from Jest to Vitest, refer to the official [Vitest Migration Guide](https://vitest.dev/guide/migration.html#jest).
 
-## Add add-ons via environment variable for testing purposes
 
-Sometimes you need to enable different configurations and enable optional components, for example, testing purposes.
-You can use the `ADDONS` environment variable to define them.
+## Jest usage
 
-```bash
-ADDONS=test-addon,test-addon2 pnpm start
+```{deprecated} Volto 18.TBD.TBD, current release at 18.10.1
+Volto core has migrated from Jest to Vitest for unit tests.
+Jest is now deprecated.
 ```
```

Comment:
It doesn't make sense to have Jest stuff here, because when this PR gets merged, we're going all in with Vitest. What's the point of discussing Jest here? I think this content should be moved into `development/add-ons/test-add-ons-18.md` under `Jest for Volto add-ons`. We'll need to do another review. @sneridagh let's make sure we're on the same (documentation) page.